cairo/ethereum/cancun/vm/instructions/system.cairo

```diff
@@ -134,25 +134,27 @@ func generic_call{
 
     let is_static = bool(is_staticcall.value + evm.value.message.value.is_static.value);
 
-    // TODO: this could be optimized using a non-copy mechanism.
-    let (accessed_addresses_copy_start, accessed_addresses_copy) = dict_copy(
-        cast(evm.value.accessed_addresses.value.dict_ptr_start, DictAccess*),
-        cast(evm.value.accessed_addresses.value.dict_ptr, DictAccess*),
-    );
-    let (accessed_storage_keys_copy_start, accessed_storage_keys_copy) = dict_copy(
-        cast(evm.value.accessed_storage_keys.value.dict_ptr_start, DictAccess*),
-        cast(evm.value.accessed_storage_keys.value.dict_ptr, DictAccess*),
-    );
+    // Fork the accessed_addresses dict segment
+    local new_dict_ptr: DictAccess*;
```

Comment:
the comment said `this could be optimized using a non-copy mechanism.`
There is a copy of the dict segment.
dict_copy use a memcpy and update the tracker in the hint and the copy_dict_segment only update the tracker, right?
So the cairo memory is not copied. Is this what it is expected?

Reply:
> dict_copy use a memcpy and update the tracker in the hint and the copy_dict_segment only update the tracker, right?
So the cairo memory is not copied. Is this what it is expected?

Yes, exactly - this is the expected behavior. The segment is not copied. But the naming is terrible and confusing. I'll update it.

Reply:
Done. removed the dict_copy hint as per rationale in https://github.com/kkrt-labs/keth/issues/863. Modified the name of the remaining hint for more clarity